doc/docs/reference/inter-workload-dependencies.md

```diff
@@ -0,0 +1,139 @@
+
+## Explicit inter-workload dependencies
+
+Ankaios supports the following dependency types:
+
+| Dependency type |  AddCondition         | meaning                                       |
+| --------------- | --------------------- | --------------------------------------------- |
+| running         | ADD_COND_RUNNING      | The dependency must be operational.           |
+| succeeded       | ADD_COND_SUCCEEDED    | The dependency must exit successfully.        |
+| failed          | ADD_COND_FAILED       | The dependency must fail.                     |
+
+The user configures the `AddCondition` for each dependency in the `dependencies` field to define one or multiple dependencies for a workload.
+
+```yaml
+apiVersion: v0.1
+workloads:
+  logger:
+    agent: agent_A
+    runtime: podman
+    dependencies:
+      storage_provider: ADD_COND_RUNNING
+    ...
+```
+
+When the `storage_provider` is operational, Ankaios starts the `logger` workload. The ExecutionState of the workload remains `Pending(WaitingToStart)` until all dependencies are met.
+
+!!! Note
+
+    Ankaios rejects manifests and workload configurations with cyclic dependencies. A manifest is valid only when its workloads and dependencies form a directed acyclic graph.
+
+This example demonstrates how to use dependency types to configure inter-workload dependencies:
+
+```mermaid
+---
+title:
+---
+flowchart RL
+    logger(logger)
+    init(init_storage)
+    storage(storage_provider)
+    err_handler(error_handler)
+
+
+    logger-- running -->storage
+    err_handler-- failed -->storage
+    storage-- succeeded -->init
+```
+
+The logging service requires an operational storage provider to write logs. Therefore, the storage provider must be started first and its initialization (init_storage) must be completed before starting the provider itself. In case of a failure, an error handler is started to manage errors.
+
+The Ankaios manifest below includes the configuration of each workload and its dependencies:
+
+```yaml linenums="1" hl_lines="6 7 15 16 31 32"
+apiVersion: v0.1
+workloads:
+  logger:
+    runtime: podman
+    agent: agent_A
+    dependencies:
+      storage_provider: ADD_COND_RUNNING # (1)
+    runtimeConfig: |
+      image: alpine:latest
+      commandOptions: [ "--entrypoint", "/bin/sleep" ]
+      commandArgs: [ "3" ]
+  storage_provider:
+    runtime: podman
+    agent: agent_B
+    dependencies:
+      init_storage: ADD_COND_SUCCEEDED # (2)
+    runtimeConfig: |
+      image: alpine:latest
+      commandOptions: [ "--entrypoint", "/bin/sh" ]
+      commandArgs: [ "-c", "sleep 5; exit 1" ]
+  init_storage: # (3)
+    runtime: podman
+    agent: agent_B
+    runtimeConfig: |
+      image: alpine:latest
+      commandOptions: [ "--entrypoint", "/bin/sleep" ]
+      commandArgs: [ "2" ]
+  error_handler:
+    runtime: podman
+    agent: agent_A
+    dependencies:
+      storage_provider: ADD_COND_FAILED # (4)
```

Comment:
```suggestion
      storage_provider: ADD_COND_RUNNING # (1)!
    runtimeConfig: |
      image: alpine:latest
      commandOptions: [ "--entrypoint", "/bin/sleep" ]
      commandArgs: [ "3" ]
  storage_provider:
    runtime: podman
    agent: agent_B
    dependencies:
      init_storage: ADD_COND_SUCCEEDED # (2)!
    runtimeConfig: |
      image: alpine:latest
      commandOptions: [ "--entrypoint", "/bin/sh" ]
      commandArgs: [ "-c", "sleep 5; exit 1" ]
  init_storage: # (3)!
    runtime: podman
    agent: agent_B
    runtimeConfig: |
      image: alpine:latest
      commandOptions: [ "--entrypoint", "/bin/sleep" ]
      commandArgs: [ "2" ]
  error_handler:
    runtime: podman
    agent: agent_A
    dependencies:
      storage_provider: ADD_COND_FAILED # (4)!
```

Without this change the block cannot be copied. The `#`  needs to excluded (see https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#stripping-comments)

Reply:
It looks better without the comments. The feature is very helpful. Thanks. However, I think it is still marked as experimental in the latest doc, but was added in version 8.5.0 already (and in the release notes of 8.5.0 it was not marked as experimental and sounds like a real feature). I guess its fine.

Reply:
We had this problem before and Ankaios will not accept YAML with a trailing `#`.